src/main/kotlin/com/expedia/graphql/schema/SchemaGenerator.kt

```diff
@@ -270,6 +271,12 @@ internal class SchemaGenerator(
         return builder.build()
     }
 
+    private fun abstractType(kClass: KClass<*>): GraphQLType = if (kClass.declaredMemberFunctions.isEmpty() && kClass.declaredMemberProperties.isEmpty()) {
```

Comment:
I think it's easier to read is the type detection logic stays in the `getGraphQLType` method.